app/javascript/components/TopicSectionSuggestion/index.jsx

```diff
@@ -1,58 +1,58 @@
-import React, {useState} from "react"
+import React, {useState, useEffect} from "react"
 import TopicPageContent from "../TopicPageContent"
 import Button from '../Button'
 import "./style.scss"
 
 const useStateWithLocalStorage = localStorageKey => {
-  const [resultsStored, setResultsStored] = React.useState(
+  const [resultsStored, setResultsStored] = useState(
     localStorage.getItem(localStorageKey) || ''
   );
 
-  React.useEffect(() => {
+  useEffect(() => {
     localStorage.setItem(localStorageKey, resultsStored);
   }, [resultsStored]);
 
   return [resultsStored, setResultsStored];
 };
 
-const TopicSectionSuggestion = ({
-    healthtitle,
-    healthintro,
-    familytitle,
-    familyintro,
-    hometitle,
-    homeintro,
-    timetitle,
-    timeintro,
-    skillstitle,
-    skillsintro,
-    currentSection,
-    allComplete }) => {
+const TopicSectionSuggestion = ({ currentSection }) => {
   const family = "results_topic_family";
   const home = "results_topic_home";
   const health = "results_topic_lifestyle";
   const time = "results_topic_time";
   const skills = "results_topic_skills";
+  const children = "results_topic_children";
   const [familyResultsStored, setFamilyResultsStored] = useStateWithLocalStorage(family);
   const [homeResultsStored, setHomeResultsStored] = useStateWithLocalStorage(home);
   const [healthResultsStored, setHealthResultsStored] = useStateWithLocalStorage(health);
   const [timeResultsStored, setTimeResultsStored] = useStateWithLocalStorage(time);
   const [skillsResultsStored, setSkillsResultsStored] = useStateWithLocalStorage(skills);
-  var amountComplete = 0;
-  if(healthResultsStored.length) { amountComplete++; }
-  if(timeResultsStored.length) { amountComplete++; }
-  if(familyResultsStored.length) { amountComplete++; }
-  if(skillsResultsStored.length) { amountComplete++; }
-  if(homeResultsStored.length) { amountComplete++; }
+  const [childrenResultsStored, setChildrenResultsStored] = useStateWithLocalStorage(children);
+
+  let amountComplete = 0;
+
+  const arr = [healthResultsStored, timeResultsStored, familyResultsStored, skillsResultsStored, childrenResultsStored, homeResultsStored]
+
+  for (let i = 0; i < arr.length; i++) {
+    if (i.length) {
+      amountComplete++
+    }
+  }
+
     return(
         <>
-        { !healthResultsStored.length || !timeResultsStored.length || !familyResultsStored.length || !skillsResultsStored.length || !homeResultsStored.length ? (
+        { 
+        !healthResultsStored.length || !timeResultsStored.length || !familyResultsStored.length || 
+        !skillsResultsStored.length || !childrenResultsStored.length || !homeResultsStored.length ? 
```

Comment:
Looking at this again I think we can actually get rid of this OR statement - we're just checking to see whether any of the topics are incomplete, but then we're checking each one again to see which link we should show, so we shouldn't need the first check.

One thing I can't tell from looking at the code - do we show the 'View your plan' button twice when the topics are all complete? 🤔

Reply:
I tried this changes, but when I remove the first check then shows two View your plan button